### PR TITLE
*: Add privilege check for drop table.

### PR DESCRIPTION
**Pull request by shenli**

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -576,6 +577,15 @@ func (d *ddl) DropTable(ctx context.Context, ti table.Ident) (err error) {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	// Check Privilege
+	pchecker := privilege.GetPrivilegeChecker(ctx)
```
Comment (zimulala) [inline ddl/ddl.go]: pChecker or other?

Comment (shenli) [inline ddl/ddl.go]: PTAL

File: privilege/privileges/privileges.go [hunk 2]
```diff
@@ -166,6 +166,9 @@ func (p *UserPrivileges) Check(ctx context.Context, db *model.DBInfo, tbl *model
 		if len(p.User) == 0 {
 			// User current user
 			p.User = variable.GetSessionVars(ctx).User
+			if len(p.User) == 0 {
```
Comment (qiuyesuifeng) [inline privilege/privileges/privileges.go]: Why `len(p.User) == 0` return true?
Also line 167 comment is `Use current user.`

File: privilege/privileges/privileges_test.go [hunk 3]
```diff
@@ -233,6 +233,29 @@ func (t *testPrivilegeSuite) TestShowGrants(c *C) {
...
+	mustExec(c, se, `SELECT * FROM todrop;`)
+
+	_, err := se.Execute("DROP TABLE todrop;")
+	c.Assert(err, NotNil)
+
+	variable.GetSessionVars(ctx).User = "root@localhost"
+	mustExec(c, se, `GRANT Drop ON test.todrop TO  'drop'@'localhost';`)
+
+	se1 := newSession(c, t.store, t.dbName)
+	ctx1, _ := se.(context.Context)
```
Comment (qiuyesuifeng) [inline privilege/privileges/privileges_test.go]: se1.(context.Context) ?

File: privilege/privileges/privileges.go (hunk 2 above)
Comment (shenli) [inline privilege/privileges/privileges.go]: For tidb-server mode, we force user do login and auth. In this situation, len(variable.GetSessionVars(ctx).User) > 0.
But in other cases (embeded-db, unittest) we do not do login and auth. So a lot of testcase will be failed. Maybe I will remove this check latter.
I will add some comments now.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**